[Keith Tan Shi Jun] iP

KT27Learn opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen (source)

DukePro frees your mind of having to remember things you need to do. It's :

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is:

1. download it from [here](https://github.com/KT27Learn/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features:

- [x]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
 
## For Developers:
If you want to clone the repository and run it on your CLI instead of using the jar file, you can follow the instructions here.

From the root of the repository:

    cd src/main/java


Compile all the classes in the package:

    javac -classpath . duke/Duke.java
  

Finally, run the program:

    java duke/Duke

keifutan20 committed:
Level-1

keifutan20 committed:
Level-2

keifutan20 committed:
Level-3

keifutan20 committed:
Fixed List starting with 0 instead of 1

keifutan20 committed:
Fixed list number bug

keifutan20 committed:
Level-3

keifutan20 committed:
Level-4

keifutan20 committed:
Level-4

keifutan20 committed:
Level-4

keifutan20 committed:
A-TextUiTesting

keifutan20 committed:
Level-5

keifutan20 committed:
Level-6

keifutan20 committed:
Level-6

keifutan20 committed:
Level-5

keifutan20 committed:
Level-6

keifutan20 committed:
A-Enums

keifutan20 committed:
Level-7

keifutan20 committed:
Level-7 Store bug fix v1

keifutan20 committed:
Level-7 Store bug v1.1

keifutan20 committed:
Level-8

KT27Learn committed:
Merge pull request #1 from KT27Learn/branch-Level-7

Branch level 7

KT27Learn committed:
Revert "Branch level 7"

KT27Learn committed:
Merge pull request #2 from KT27Learn/revert-1-branch-Level-7

Revert "Branch level 7"

keifutan20 committed:
Merge branch-Level-8

keifutan20 committed:
Revert to A-Enums Commit

Made an error in merging branches and hence reverting to a previous commit
to allow for the correction to be made

keifutan20 committed:
Revert to Enums Commit

Previous commit failed to remove the data folder

keifutan20 committed:
Add level-7

keifutan20 committed:
Add level-7

keifutan20 committed:
Level-8

Added updates for Level-8 including date and times

keifutan20 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8

# Conflicts:
#	data/duke.txt
#	src/main/java/Command.java
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/DukeException.java
#	src/main/java/Event.java
#	src/main/java/Store.java
#	src/main/java/Task.java
#	src/main/java/Tasklist.java
#	src/main/java/ToDo.java
#	text-ui-test/EXPECTED.TXT
#	text-ui-test/input.txt

keifutan20 committed:
A-MoreOOP

Refactored code to extract out closely related code as classes

keifutan20 committed:
A-Packages: Organise Code into packages

Organize the classes into suitable java packages.

keifutan20 committed:
A-JUnit: Add JUnit tests

Added JUnit test to test the behaviour of the code

keifutan20 committed:
A-JUnit: Add JUnit Tests

Add JUnit tests to test the behaviour of the code

keifutan20 committed:
A-Jar: Create a JAR file

Packaged the app as an executable JAR file

keifutan20 committed:
A-JavaDoc: Add JavaDoc comments

Add header comments to classes and methods

keifutan20 committed:
A-CodingStandard: Follow the Coding Standard

Tweak the code to comply with a coding standard

keifutan20 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard: Follow the Coding Standard

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/HelpCommand.java
#	src/main/java/duke/command/InvalidCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/exception/IndexOutOfRangeException.java
#	src/main/java/duke/exception/InvalidArgumentException.java
#	src/main/java/duke/exception/InvalidCommandException.java
#	src/main/java/duke/exception/MissingArgumentException.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/ToDo.java
#	src/main/java/duke/util/Duke.java
#	src/main/java/duke/util/DukeDate.java
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/Store.java
#	src/main/java/duke/util/Tasklist.java
#	src/main/java/duke/util/Ui.java

keifutan20 committed:
Level-9: Filter tasks by searching for a keyword

Give users a way to find a task by searching for a keyword

keifutan20 committed:
Moved Duke out of duke.util

Moved Duke out of duke.util to prep for Gradle

keifutan20 committed:
Add Gradle Support

A-Gradle. Add Gradle Support to Duke

keifutan20 committed:
Added build.gradle file

Added build.gradle file that was initially left out

keifutan20 committed:
Add Checkstyle. Change code to fit style guide

Modified code to suit the style guide. Implemented checkstyle
into gradle.

keifutan20 committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add Checkstyle. Change code to fit style guide

# Conflicts:
#	build.gradle

keifutan20 committed:
Add a GUI to Duke using JavaFX

Level-10. Added a GUI to Duke using JavaFX technology.

yeo-yiheng commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,32 @@
+
+import java.io.IOException;
+
+import duke.ui.MainWindow;
```
Perhaps you could separate the different import statements based on their packages?

IceWizard4902 replied:
Better separation between the import should be nice

yeo-yiheng commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public void start(Stage stage) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
```
I may be wrong on this but the class `FXMLLoader` seems like it should be named `FxmlLoader`?

KT27Learn replied:
Hi I have checked and believed that FXMLLoader is the correct one in this case as upon checking the class name is indeed declared as FXMLLoader in the javafx package

yeo-yiheng commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,43 @@
+public class DeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
+    private Tasklist list;
+    private final int toDelete;
```
Perhaps your final variable should be in all uppercase?

IceWizard4902 replied:
Should the toDelete be in all caps (TODELETE)?

KT27Learn replied:
Thanks for raising this up, I believe I had missed out on this. I will rectify it in a future commit.

yeo-yiheng commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,45 @@
+public class MarkCommand extends Command {
+
+    public static final String COMMAND_WORD = "add";
+    private final int toMark;
```
Perhaps the final variable count be all uppercase? I have noticed this in some other areas as well.

IceWizard4902 replied:
I think you may forget to change the final variable to all uppercase.

KT27Learn replied:
Thanks for raising this up, I believe I had missed out on this. I will rectify it in a future commit.

yeo-yiheng commented on src/main/java/duke/ui/MainWindow.java:
```diff
@@ -0,0 +1,74 @@
+package duke.ui;
+
+import duke.Duke;
```
Perhaps the import statements can be grouped by their packages? I noticed this in other areas as well.

yeo-yiheng reviewed (commented):
LGTM. I like that your program is very neatly arranged and written. Just some really minor styling to change and we're good to go!

IceWizard4902 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+     * Returns a boolean of whether current command is the 'bye' command
+     *
+     * @return boolean returns boolean of whether current command is 'bye' command
+     */
```
Perhaps the name of the method can be changed to isByeCommand to improve readability

IceWizard4902 reviewed (approved):
Everything looks great, and I think that it is an overall job well done! There are no major issues, and the code base was easy to understand. There are just some small pointers that we have both brought up.

keifutan20 committed:
Add refinements based off peer feedback in PR Review

keifutan20 committed:
Add assertions to document important assumptions

Previously there was a few methods which contained if-else
statements and switch statements that made assumptions on the current
programme state

Having assertions would create a safety net to ensure the assumptions
hold

Let's put assertions for the following situations:
* Internal Invariants
* Control-Flow Invariants
* Preconditions
* Postconditions
* Class Invariants

KT27Learn committed:
Merge pull request #3 from KT27Learn/branch-A-Assertions

Add assertions to document important assumptions

keifutan20 committed:
Improve Code Quality

There were portions of the code that did not adhere to the
5.4 Code Quality guidelines in the textbook.

Not adhering to the guidelines will result in unreadable code
and cause confusion in readers.

KT27Learn committed:
Merge pull request #4 from KT27Learn/branch-A-CodeQuality

Improve Code Quality

keifutan20 committed:
Improve Code Quality

There were portions of the code that did not adhere to the
5.4 Code Quality guidelines in the textbook.

Not adhering to the guidelines will result in unreadable code
and cause confusion in readers.

keifutan20 committed:
Use streams within code

KT27Learn committed:
Merge pull request #5 from KT27Learn/branch-A-Streams

Use streams within code

keifutan20 committed:
Set up continuous integration using github actions

keifutan20 committed:
Fix Assertion bug

keifutan20 committed:
Fix assertion bug

keifutan20 committed:
Fix store path

keifutan20 committed:
Remove UiTest

keifutan20 committed:
Add update functionality to Duke

KT27Learn committed:
Merge branch 'master' into branch-C-Update

KT27Learn committed:
Merge pull request #6 from KT27Learn/branch-C-Update

Add update functionality to Duke

keifutan20 committed:
Improve GUI

keifutan20 committed:
Improve code style

KT27Learn committed:
Merge pull request #8 from KT27Learn/branch-A-BetterGui

Improve GUI

keifutan20 committed:
Update find function bug

keifutan20 committed:
Update find to become case-insensitive

keifutan20 committed:
Update user guide

KT27Learn committed:
Set theme jekyll-theme-cayman

KT27Learn committed:
Update user guide bugs

keifutan20 committed:
Update Readme bugs

keifutan20 committed:
Update readme style bugs

keifutan20 committed:
Fix Duke gui text bug

keifutan20 committed:
Update GUI styles
